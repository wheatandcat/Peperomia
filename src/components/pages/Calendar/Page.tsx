import React, { memo, useCallback } from 'react';
import {
  View,
  FlatList,
  ListRenderItemInfo,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import CreateHeader from 'components/organisms/Header/CreateHeader.tsx';
import Header from 'components/organisms/Header/Header.tsx';
import { getKindData } from 'lib/kind';
import Card from 'components/organisms/Calendar/Card.tsx';
import Add from 'components/organisms/Calendar/Add.tsx';
import theme from 'config/theme';
import ItemWrap from 'components/organisms/ItemWrap/ItemWrap';
import { ConnectedType, CalendarType, ItemDetailType } from './Connected';

dayjs.extend(advancedFormat);

type Props = ConnectedType & {
  calendar: CalendarType;
};

type ItemDetailType2 = ItemDetailType & {
  add?: boolean;
};

type Item = {
  item1: ItemDetailType2;
  item2: ItemDetailType2;
};

const CalendarPage: React.FC<Props> = (props) => {
  const calendar = props.calendar;
  const itemDetails = calendar.item.itemDetails || [];

  const config = getKindData(calendar.item.kind || '');

  const renderItem = useCallback(
    ({ item }: ListRenderItemInfo<Item>) => {
      return (
        <View style={styles.cards}>
          {(() => {
            if (item?.item1?.add) {
              return (
                <TouchableOpacity
                  onPress={props.onAddItemDetail}
                  style={styles.card1}
                >
                  <View>
                    <Add />
                  </View>
                </TouchableOpacity>
              );
            }

            if (item.item1) {
              return (
                <TouchableOpacity
                  onPress={() => props.onItemDetail(item.item1.id)}
                  style={styles.card1}
                >
                  <Card {...item.item1} />
                </TouchableOpacity>
              );
            }

            return null;
          })()}
          {(() => {
            if (item?.item2?.add) {
              return (
                <TouchableOpacity
                  onPress={props.onAddItemDetail}
                  style={styles.card2}
                >
                  <View>
                    <Add />
                  </View>
                </TouchableOpacity>
              );
            }

            if (item.item2) {
              return (
                <TouchableOpacity
                  onPress={() => props.onItemDetail(item.item2.id)}
                  style={styles.card2}
                >
                  <Card {...item.item2} />
                </TouchableOpacity>
              );
            }

            return null;
          })()}
        </View>
      );
    },
    [props]
  );

  const listItem = [
    ...itemDetails.slice(1),
    {
      add: true,
      id: 'add-item',
      title: '',
      kind: '',
    },
  ];

  const list1 = listItem.filter((_, index) => {
    return index % 2 === 0;
  });
  const list2 = listItem.filter((_, index) => {
    return index % 2 === 1;
  });

  const data = list1.map((_, index) => ({
    item1: list1[index] || null,
    item2: list2[index] || null,
  })) as Item[];

  return (
    <ItemWrap kind={calendar.item.kind || ''}>
      <FlatList<Item>
        keyExtractor={(_, index) => `card_${index}`}
        data={data}
        renderItem={renderItem}
        ListHeaderComponent={
          <View style={styles.header}>
            {props.create ? (
              <CreateHeader
                color={config.backgroundColor}
                date={dayjs(calendar.date).format('YYYY年MM月DD日')}
                title={calendar.item.title || ''}
                kind={calendar.item.kind || ''}
                onClose={props.onDismiss}
              />
            ) : (
              <Header
                color={config.backgroundColor}
                date={dayjs(calendar.date).format('YYYY年MM月DD日')}
                title={calendar.item.title || ''}
                kind={calendar.item.kind || ''}
                public={calendar.public}
                onClose={props.onDismiss}
                onUpdate={props.onUpdate}
                onShare={props.onShare}
                onDelete={props.onDelete}
              />
            )}
          </View>
        }
        ListFooterComponent={<View style={styles.footer} />}
        showsVerticalScrollIndicator={false}
        showsHorizontalScrollIndicator={false}
        contentContainerStyle={styles.contentContainerStyle}
      />
    </ItemWrap>
  );
};

export default memo(CalendarPage);

const styles = StyleSheet.create({
  header: {
    paddingBottom: theme().space(2),
  },
  footer: {
    width: '100%',
    height: '100%',
    backgroundColor: theme().color.white,
  },
  cards: {
    width: '100%',
    flexDirection: 'row',
    backgroundColor: theme().color.white,
  },
  card1: {
    width: '50%',
    padding: theme().space(2),
    paddingRight: theme().space(2),
  },
  card2: {
    width: '50%',
    padding: theme().space(2),
    paddingRight: theme().space(2),
  },
  contentContainerStyle: {
    flex: 1,
  },
});
