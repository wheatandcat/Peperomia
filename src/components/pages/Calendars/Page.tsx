import React, { Component } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  ActivityIndicator,
  Animated,
  SafeAreaView,
  ScrollView,
  StatusBar,
} from 'react-native';
import { Calendar, LocaleConfig, CalendarTheme } from 'react-native-calendars';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import GestureRecognizer from 'react-native-swipe-gestures';
import EStyleSheet from 'react-native-extended-stylesheet';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { SelectCalendar } from 'domain/calendar';
import theme from 'config/theme';
import ImageDay from 'components/organisms/Calendars/Image';
import DayText from 'components/organisms/Calendars/DayText';
import GlobalStyles from '../../../GlobalStyles';
import { ConnectedType } from './Connected';

dayjs.extend(advancedFormat);

const AnimatedSafeAreaView = Animated.createAnimatedComponent(SafeAreaView);

LocaleConfig.locales.jp = {
  monthNames: [
    '1月',
    '2月',
    '3月',
    '4月',
    '5月',
    '6月',
    '7月',
    '8月',
    '9月',
    '10月',
    '11月',
    '12月',
  ],
  monthNamesShort: [
    '1月',
    '2月',
    '3月',
    '4月',
    '5月',
    '6月',
    '7月',
    '8月',
    '9月',
    '10月',
    '11月',
    '12月',
  ],
  dayNames: ['日', '月', '火', '水', '木', '金', '土'],
  dayNamesShort: ['日', '月', '火', '水', '木', '金', '土'],
};
LocaleConfig.defaultLocale = 'jp';

type Props = ConnectedType & {
  loading: boolean;
  calendars: SelectCalendar[];
};

type State = {
  currentDate: string;
  count: number;
  backgroundColor: any;
};

const backgroundColors = [
  '#FDDACB',
  '#FBEDF2',
  '#DCF6FA',
  '#fdeff2',
  '#FFE8AE',
  '#7BB07D',
  '#a0d8ef',
  '#f2f2b0',
  theme().color.beige,
  '#E3E3FE',
  '#FFE7D0',
  '#BFD3B7',
];

const config = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80,
};

export default class extends Component<Props, State> {
  state = {
    currentDate: dayjs().format('YYYY-MM-01'),
    count: 0,
    backgroundColor: new Animated.Value(dayjs().month()),
  };

  calendar: any;
  componentDidMount() {
    const currentMonth = dayjs(this.state.currentDate).month();

    Animated.timing(this.state.backgroundColor, {
      toValue: currentMonth,
      useNativeDriver: false,
    }).start();
  }

  onNextMonth = () => {
    const count = this.state.count + 1;
    const currentMonth = dayjs(this.state.currentDate)
      .add(count, 'month')
      .month();

    this.setState({
      count,
    });

    if (currentMonth === 0) {
      Animated.timing(this.state.backgroundColor, {
        toValue: 12,
        useNativeDriver: false,
      }).start((props) => {
        if (props.finished) {
          this.setState({
            backgroundColor: new Animated.Value(0),
          });
        }
      });
    } else {
      Animated.timing(this.state.backgroundColor, {
        toValue: currentMonth,
        useNativeDriver: false,
      }).start();
    }
    this.calendar.addMonth(1);
  };

  onPrevMonth = () => {
    const count = this.state.count - 1;
    const currentMonth = dayjs(this.state.currentDate)
      .add(count, 'month')
      .month();

    this.setState({
      count,
    });

    if (currentMonth === 11) {
      Animated.timing(this.state.backgroundColor, {
        toValue: -1,
        useNativeDriver: false,
      }).start((props) => {
        if (props.finished) {
          this.setState({
            backgroundColor: new Animated.Value(12),
          });
        }
      });
    } else {
      Animated.timing(this.state.backgroundColor, {
        toValue: currentMonth,
        useNativeDriver: false,
      }).start();
    }
    this.calendar.addMonth(-1);
  };

  render() {
    const backgroundColor = this.state.backgroundColor.interpolate({
      inputRange: [-1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
      outputRange: [
        backgroundColors[11],
        ...backgroundColors,
        backgroundColors[0],
      ],
    });

    const currentMonth = dayjs(this.state.currentDate)
      .add(this.state.count, 'month')
      .month();

    const currentDate = dayjs(this.state.currentDate)
      .add(this.state.count, 'month')
      .format('YYYY-MM-DD');

    const animationStyle = {
      backgroundColor,
    };

    return (
      <ScrollView
        style={[
          styles.scroll,
          {
            backgroundColor: backgroundColors[currentMonth],
          },
        ]}
      >
        <StatusBar
          backgroundColor={theme().color.white}
          barStyle={'dark-content'}
        />
        <AnimatedSafeAreaView
          style={[GlobalStyles.droidSafeArea, styles.safeArea, animationStyle]}
        />
        <GestureRecognizer
          onSwipeLeft={this.onNextMonth}
          onSwipeRight={this.onPrevMonth}
          config={config}
        >
          <AnimatedSafeAreaView style={[styles.root, animationStyle]}>
            <View style={styles.headerContainer}>
              <TouchableOpacity onPress={this.onPrevMonth}>
                <MaterialCommunityIcons
                  name="chevron-left"
                  size={30}
                  color={theme().color.main}
                />
              </TouchableOpacity>

              <View style={styles.yearContainer}>
                <Text style={styles.year}>
                  {dayjs(this.state.currentDate)
                    .add(this.state.count, 'month')
                    .format('YYYY年MM月')}
                </Text>
              </View>

              <TouchableOpacity onPress={this.onNextMonth}>
                <MaterialCommunityIcons
                  name="chevron-right"
                  size={30}
                  color={theme().color.main}
                />
              </TouchableOpacity>
            </View>

            <View style={styles.weekNameContainer}>
              <Text style={[styles.weekText, { color: theme().color.red }]}>
                日
              </Text>
              <Text style={styles.weekText}>月</Text>
              <Text style={styles.weekText}>火</Text>
              <Text style={styles.weekText}>水</Text>
              <Text style={styles.weekText}>木</Text>
              <Text style={styles.weekText}>金</Text>
              <Text style={[styles.weekText, { color: theme().color.sky }]}>
                土
              </Text>
            </View>
            {this.props.loading ? (
              <ActivityIndicator size="large" color={theme().mode.text} />
            ) : (
              <Calendar
                ref={(ref) => (this.calendar = ref)}
                style={{
                  backgroundColor: backgroundColors[currentMonth],
                }}
                monthFormat={''}
                hideDayNames
                hideArrows
                theme={calendarTheme}
                dayComponent={({ date }) => {
                  const schedule = this.props.calendars.find(
                    (item) => item.date === date.dateString
                  );

                  if (schedule) {
                    return (
                      <TouchableOpacity
                        onPress={() =>
                          this.props.onSchedule(schedule.itemId, schedule.title)
                        }
                      >
                        <ImageDay
                          currentDate={currentDate}
                          kind={schedule.kind}
                          day={String(date.day)}
                        />
                      </TouchableOpacity>
                    );
                  }

                  return (
                    <TouchableOpacity
                      onPress={() => this.props.onCreate(date.dateString)}
                    >
                      <DayText
                        currentDate={currentDate}
                        color={backgroundColors[currentMonth]}
                        day={date.day}
                      />
                    </TouchableOpacity>
                  );
                }}
              />
            )}
            <View style={styles.calendarBottom} />
          </AnimatedSafeAreaView>
        </GestureRecognizer>
      </ScrollView>
    );
  }
}

const calendarTheme: CalendarTheme = {
  'stylesheet.calendar.main': {
    week: {
      marginTop: 0,
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
  },
  textMonthFontSize: 18,
  monthTextColor: '#000',
  textMonthFontWeight: '600',
  'stylesheet.calendar.header': {
    header: {
      height: 0,
    },
    dayHeader: {
      fontWeight: '600',
      paddingBottom: 10,
      color: '#000',
    },
  },
};

const styles = EStyleSheet.create({
  root: {
    flex: 1,
    height: '100%',
    width: '100%',
  },
  headerContainer: {
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
  },
  yearContainer: {
    paddingVertical: 20,
    justifyContent: 'center',
  },
  year: {
    textAlign: 'center',
    color: theme().color.darkGray,
    fontSize: 25,
    fontWeight: '600',
  },
  weekNameContainer: {
    marginTop: 7,
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginHorizontal: 4,
  },
  weekText: {
    paddingVertical: 12,
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 18,
    fontWeight: '600',
  },
  calendarBottom: {
    borderColor: 'gray',
    marginHorizontal: 4,
    borderBottomWidth: 1,
  },
  scroll: {
    height: '100%',
    width: '100%',
  },
  safeArea: {
    flex: 0,
  },
});
