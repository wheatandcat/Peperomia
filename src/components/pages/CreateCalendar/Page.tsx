import React, { memo } from 'react';
import CreateCalendar from 'components/templates/CreateCalendar/Page';
import { QueryProps } from './Plain';
import { ConnectedType } from './Connected';

type Props = ConnectedType & {
  mutationData: QueryProps['mutationData'];
};

const CreateCalendarPage: React.FC<Props> = (props) => {
  return <CreateCalendar loading={false} {...props} calendar={null} />;
};

export default memo(CreateCalendarPage);
