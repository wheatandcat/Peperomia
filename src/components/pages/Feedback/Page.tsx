import React, { Component } from 'react';
import {
  View,
  Text,
  TextInput,
  KeyboardAvoidingView,
  Image,
  Platform,
  InputAccessoryView,
  Keyboard,
  StyleSheet,
} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import { Button, Overlay } from 'react-native-elements';
import theme, { darkMode } from 'config/theme';

type Props = {
  isOpen: boolean;
  loading: boolean;
  onFeedback: (message: string) => void;
  onClose: () => void;
};

type State = {
  message: string;
};

export default class extends Component<Props, State> {
  state = {
    message: '',
  };

  render() {
    const inputAccessoryViewID = 'feedbackID';

    return (
      <>
        <Overlay isVisible={this.props.isOpen}>
          <View style={estyles.dialogContainer}>
            <View style={styles.imageContainer}>
              <Image
                source={
                  darkMode()
                    ? require('../../../img/check_dark.png')
                    : require('../../../img/check.png')
                }
                style={styles.image}
              />
            </View>
            <Text style={estyles.dialogText}>
              フィードバックありがとうございます！
            </Text>
            <View style={styles.imageContainer}>
              <Button
                title="閉じる"
                containerStyle={styles.button}
                onPress={this.props.onClose}
              />
            </View>
          </View>
        </Overlay>
        <View style={estyles.root}>
          <KeyboardAvoidingView behavior="padding">
            <View>
              <View style={estyles.description}>
                <Text style={estyles.descriptionText}>
                  気になる点や不具合等があれば、こちらからフィードバックお願いします。
                </Text>
              </View>
              <View style={estyles.textInputContainer}>
                <TextInput
                  placeholder="フィードバックを書く"
                  placeholderTextColor={theme().mode.secondaryText}
                  multiline
                  style={styles.textInput}
                  onChangeText={(message) => {
                    this.setState({
                      message,
                    });
                  }}
                  testID="inputTextFeedbackMemo"
                  inputAccessoryViewID={inputAccessoryViewID}
                  autoFocus
                >
                  <Text style={estyles.inputText}>{this.state.message}</Text>
                </TextInput>
              </View>
              <Button
                title="フィードバックを送信する"
                titleStyle={styles.buttonTitle}
                buttonStyle={styles.sendButton}
                disabled={this.state.message === '' || this.props.loading}
                onPress={() => this.props.onFeedback(this.state.message)}
                loading={this.props.loading}
              />
            </View>
          </KeyboardAvoidingView>

          {Platform.OS === 'ios' && (
            <InputAccessoryView nativeID={inputAccessoryViewID}>
              <View style={styles.keyboardButtonContainer}>
                <Button
                  buttonStyle={styles.keyboardButton}
                  onPress={() => Keyboard.dismiss()}
                  title="閉じる"
                />
              </View>
            </InputAccessoryView>
          )}
        </View>
      </>
    );
  }
}

const estyles = EStyleSheet.create({
  root: {
    backgroundColor: '$background',
    height: '100%',
  },
  description: {
    backgroundColor: '$secondaryBackground',
  },
  descriptionText: {
    paddingHorizontal: theme().space(3),
    paddingVertical: theme().space(2),
    fontSize: 16,
    lineHeight: 20,
    color: '$text',
  },
  textInputContainer: {
    paddingTop: theme().space(1),
    paddingHorizontal: theme().space(3),
    height: 250,
    backgroundColor: '$background',
  },
  inputText: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '400',
    color: '$text',
  },
  dialogContainer: {
    backgroundColor: '$background',
    height: 280,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dialogText: {
    fontSize: 15,
    fontWeight: '600',
    paddingVertical: theme().space(2),
    paddingHorizontal: theme().space(1),
    color: '$text',
  },
});

const styles = StyleSheet.create({
  imageContainer: {
    padding: theme().space(3),
  },
  image: {
    height: 125,
    width: 125,
  },
  button: {
    width: 120,
  },
  textInput: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '400',
  },
  buttonTitle: {
    fontWeight: '600',
  },
  sendButton: {
    borderRadius: 0,
    paddingVertical: theme().space(3),
  },
  keyboardButtonContainer: {
    alignItems: 'flex-end',
  },
  keyboardButton: {
    width: 80,
    right: 0,
    borderRadius: 0,
  },
});
