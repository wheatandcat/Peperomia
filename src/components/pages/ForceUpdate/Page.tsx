import React, { useCallback } from 'react';
import { View, ScrollView, Text } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import AppLink from 'react-native-app-link';
import { Button } from 'react-native-elements';

type Props = {};

const ForceUpdate: React.FC<Props> = () => {
  const onPress = useCallback(() => {
    const appStoreId = 1460583871;
    const playStoreId = 'com.wheatandcat.peperomia';
    AppLink.openInStore({
      appName: 'シェアフル',
      appStoreId,
      appStoreLocale: 'jp',
      playStoreId,
    });
  }, []);

  return (
    <View style={styles.root}>
      <ScrollView>
        <View style={styles.inner}>
          <View style={styles.textContainer}>
            <Text style={styles.title}>
              最新のバージョンのアプリを{'\n'}インストールしてください。
            </Text>
          </View>
          <View style={styles.buttonContainer}>
            <Button title="ストアへ移動する" onPress={onPress} />
          </View>
        </View>
      </ScrollView>
    </View>
  );
};

export default ForceUpdate;

const styles = EStyleSheet.create({
  root: {
    backgroundColor: '$background',
    height: '100%',
  },
  inner: {
    height: '100%',
    paddingHorizontal: 15,
    paddingVertical: 15,
  },
  title: {
    color: '$text',
    fontSize: 16,
    fontWeight: 'bold',
    textAlign: 'center',
  },

  textContainer: {
    paddingTop: 15,
    paddingBottom: 15,
  },
  buttonContainer: {
    paddingHorizontal: 15,
    paddingVertical: 15,
  },
});
