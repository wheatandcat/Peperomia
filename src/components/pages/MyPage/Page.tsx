import React, { FC } from 'react';
import { View, ScrollView, Text, Linking, StyleSheet } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import { ListItem, Button, Divider } from 'react-native-elements';
import { MaterialIcons } from '@expo/vector-icons';
import Spinner from 'react-native-loading-spinner-overlay';
import Constants from 'expo-constants';
import theme from 'config/theme';

type Props = {
  loading: boolean;
  LoadingText: string;
  email: string;
  onBackup: () => void;
  onNotificationSetting: () => Promise<void>;
};

const url = 'https://amazing-hawking-a280c3.netlify.com/general/account/';

const MyPage: FC<Props> = (props) => (
  <View style={estyles.root}>
    <Spinner
      visible={props.loading}
      textContent={props.LoadingText}
      textStyle={{ color: theme().color.background.main }}
    />
    <ScrollView>
      <View style={styles.contents}>
        <View style={estyles.emailContainer}>
          <Text style={estyles.emailTitle}>メールアドレス:</Text>
          <Text style={estyles.email}>{props.email}</Text>
        </View>
        <View style={styles.guideContainer}>
          <Button
            title="ユーザー登録とは？"
            type="clear"
            titleStyle={styles.guide}
            icon={
              <MaterialIcons
                name="chevron-right"
                size={20}
                color={theme().color.primary.main}
              />
            }
            iconRight
            onPress={() => Linking.openURL(url)}
          />
        </View>
      </View>

      {!Constants.isDevice && (
        <>
          <Divider />
          <ListItem onPress={props.onBackup} bottomDivider>
            <ListItem.Content>
              <ListItem.Title>バックアップを作成する</ListItem.Title>
            </ListItem.Content>
            <ListItem.Chevron />
          </ListItem>
        </>
      )}

      <ListItem onPress={props.onNotificationSetting} bottomDivider>
        <ListItem.Content>
          <ListItem.Title>通知を設定する</ListItem.Title>
        </ListItem.Content>
        <ListItem.Chevron />
      </ListItem>

      <View style={styles.list}>
        <View style={styles.guideContainer}>
          <Button
            title="通知設定とは？"
            type="clear"
            titleStyle={styles.guide}
            icon={
              <MaterialIcons
                name="chevron-right"
                size={20}
                color={theme().color.primary.main}
              />
            }
            iconRight
            onPress={() => Linking.openURL(url)}
          />
        </View>
      </View>
    </ScrollView>
  </View>
);

const estyles = EStyleSheet.create({
  root: {
    backgroundColor: '$settingRoot',
    height: '100%',
  },
  emailContainer: {
    padding: theme().space(2),
    width: '100%',
    justifyContent: 'center',
    backgroundColor: '$settingMenu',
  },
  emailTitle: {
    color: '$text',
  },
  email: {
    fontSize: 20,
    fontWeight: '600',
    color: '$text',
  },
  menuContainer: {
    backgroundColor: '$settingMenu',
  },
  menuText: {
    color: '$text',
  },
});

const styles = StyleSheet.create({
  contents: {
    marginTop: theme().space(3),
    marginHorizontal: theme().space(2),
    marginBottom: theme().space(4),
  },
  list: {
    marginTop: theme().space(2),
    marginHorizontal: theme().space(2),
    marginBottom: theme().space(1),
  },
  guideContainer: {
    alignItems: 'flex-end',
  },
  guide: {
    fontSize: 14,
    color: theme().color.primary.main,
  },
});

export default MyPage;
