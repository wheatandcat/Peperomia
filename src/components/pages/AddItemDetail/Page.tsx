import React, { memo } from 'react';
import CreateCalendar from 'components/templates/CreateCalendar/Page';
import { QueryProps } from './Plain';
import { ConnectedType } from './Connected';

type Props = ConnectedType & {
  mutationData: QueryProps['mutationData'];
};

const AddItemDetailPage: React.FC<Props> = (props) => {
  return (
    <CreateCalendar
      {...props}
      loading={props.mutationData.loading}
      itemDetail={null}
    />
  );
};

export default memo(AddItemDetailPage);
