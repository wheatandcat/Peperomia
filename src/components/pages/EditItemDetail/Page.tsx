import React, { memo } from 'react';
import CreateCalendar from 'components/templates/CreateCalendar/Page';
import { ItemDetailQuery } from 'queries/api/index';
import { QueryProps } from './Plain';
import { ConnectedType } from './Connected';

export type Props = ConnectedType & {
  itemDetail: ItemDetailQuery['itemDetail'];
  mutationData: QueryProps['mutationData'];
};

const EditItemDetailPage: React.FC<Props> = (props) => {
  return <CreateCalendar {...props} loading={props.mutationData.loading} />;
};

export default memo(EditItemDetailPage);
