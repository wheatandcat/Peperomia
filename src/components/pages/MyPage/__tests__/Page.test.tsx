import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Page from '../Page';

describe('components/pages/MyPage/Page.tsx', () => {
  let wrapper: ShallowWrapper;

  const propsData: any = () => ({
    loading: false,
    LoadingText: '',
    refreshData: jest.fn(),
    email: 'test',
    onBackup: jest.fn(),
    onRestore: jest.fn(),
    onNotificationSetting: jest.fn(),
  });

  beforeEach(() => {
    wrapper = shallow(<Page {...propsData()} />);
  });

  it('正常に表示されている', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
