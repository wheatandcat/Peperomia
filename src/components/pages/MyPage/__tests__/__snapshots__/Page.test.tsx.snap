// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/MyPage/Page.tsx 正常に表示されている 1`] = `
<View>
  <Spinner
    animation="none"
    cancelable={false}
    color="white"
    overlayColor="rgba(0, 0, 0, 0.25)"
    size="large"
    textContent=""
    textStyle={
      Object {
        "color": "#fff",
      }
    }
    visible={false}
  />
  <ScrollViewMock>
    <View>
      <View>
        <Text>
          メールアドレス:
        </Text>
        <Text>
          test
        </Text>
      </View>
      <View>
        <ForwardRef(Themed.Button)
          icon={
            <Icon
              allowFontScaling={false}
              color="#006835"
              name="chevron-right"
              size={20}
            />
          }
          iconRight={true}
          onPress={[Function]}
          title="ユーザー登録とは？"
          type="clear"
        />
      </View>
    </View>
    <Themed.Divider />
    <Themed.ListItem
      bottomDivider={true}
      onPress={[MockFunction]}
      rightIcon={
        Object {
          "color": "#333631",
          "name": "chevron-right",
        }
      }
      title="通知を設定する"
    />
    <View>
      <View>
        <ForwardRef(Themed.Button)
          icon={
            <Icon
              allowFontScaling={false}
              color="#006835"
              name="chevron-right"
              size={20}
            />
          }
          iconRight={true}
          onPress={[Function]}
          title="通知設定とは？"
          type="clear"
        />
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
